hal/inc/adc_hal.h

```diff
@@ -42,6 +42,11 @@ int32_t hal_adc_read(pin_t pin);
 void hal_adc_dma_init();
 int hal_adc_sleep(bool sleep, void* reserved);
 
+
+// Deprecated *dynalib* APIs for backwards compatibility
+void inline __attribute__((always_inline, deprecated("Use hal_adc_set_sample_time() instead"))) HAL_ADC_Set_Sample_Time(uint8_t sample_time) { hal_adc_set_sample_time(sample_time); }
```

Comment:
Let's move to a separate `adc_hal_compat.h` header. Same for all the other headers.